tests/Unit/Domain/Institution/ValueObject/EmailDomainWildcardTest.php

```diff
@@ -29,6 +29,7 @@ public function test_happy_flow() : void
     {
         $domain = new EmailDomainWildcard('*.example.com');
         $this->assertInstanceOf(EmailDomainWildcard::class, $domain);
+        $this->assertEquals('*.example.com', $domain->getEmailDomain());
```

Comment:
Please consider renaming the `getEmailDomain` method as it won't return a domain.

Reply:
I find it hard to think of a better name for this method. I'm no fan of using these kind of method names as this makes the code potentially less expressive. I've considered:
1. Adding a __toString method as a remedy (again not very elegant)
1. Renaming the method to a synonym: `getDomainName`

Do you have a better suggestion?

Reply:
1. Please don't use `__toString` ever because you never know were it is used because it could be implicitly casted.
2. Maybe `getWildcardDomain` or `getCatchAllDomain`

Reply:
1. That's what I do not like about that solution
2. The collection relies on the getEmailDomain for the EmailDomainInterface implementors. Adding an additional method would make working with the domains much harder.

Reply:
1. :+1: 
2. Then the interface shouldn't be used for the `EmailDomainWildcard` object because it doesn't properly reflect the method and is used incorrectly. If you want to use it in the collection the create an interface for that (e.a. `CollectableEmailInterface`). Also I don't see the usage of `getEmailDomain` other than in a test?

Reply:
Strictly speaking your are right. The name 'EmailDomain' might not be the correct term. And might be better named someting like: EmailDomainMatcherInterface.

FYI the method is used in the Service: `\Surfnet\AzureMfa\Domain\Institution\Collection\InstitutionCollection::getByEmailDomain`

Reply:
I don't see the method `getByEmailDomain` being used only the `domainMatches` method seems to be used or am I missing something here?